joblib/numpy_pickle.py

```diff
@@ -440,33 +469,50 @@ def load(filename, mmap_mode=None):
     object might not match the original pickled object. Note that if the
     file was saved with compression, the arrays cannot be memmaped.
     """
-    with open(filename, 'rb') as file_handle:
-        # We are careful to open the file handle early and keep it open to
-        # avoid race-conditions on renames. That said, if data are stored in
-        # companion files, moving the directory will create a race when
-        # joblib tries to access the companion files.
-        if _read_magic(file_handle) == _ZFILE_PREFIX:
-            if mmap_mode is not None:
-                warnings.warn('file "%(filename)s" appears to be a zip, '
-                              'ignoring mmap_mode "%(mmap_mode)s" flag passed'
-                              % locals(), Warning, stacklevel=2)
-            unpickler = ZipNumpyUnpickler(filename, file_handle=file_handle)
-        else:
-            unpickler = NumpyUnpickler(filename, file_handle=file_handle,
-                                       mmap_mode=mmap_mode)
-
-        try:
-            obj = unpickler.load()
-        except UnicodeDecodeError as exc:
-            # More user-friendly error message
-            if PY3_OR_LATER:
-                new_exc = ValueError(
-                    'You may be trying to read with '
-                    'python 3 a joblib pickle generated with python 2. '
-                    'This feature is not supported by joblib.')
-                new_exc.__cause__ = exc
-                raise new_exc
-        finally:
-            if hasattr(unpickler, 'file_handle'):
-                unpickler.file_handle.close()
-        return obj
+    # Backward compatibility with old compression strategy
+    with open(filename, 'rb') as f:
+        compressor = _detect_compressor(f)
+        if compressor == 'compat':
+            f.close()
```

Comment:
Just curious, why the `f.close()` here? I would have thought the file would be closed when existing the `with` block anyway.